src/javascripts/components/number.js

```diff
@@ -16,18 +16,26 @@ export default class Number extends React.Component {
     format: "0,0[.][00]",
   })
 
-  _formatNumber() {
-    if (!this.props.format) return
+  _formatNumber(currentNumber) {
+    if (!this.props.format) return currentNumber
 
-    let currentNumber = this._toNumeral(this.props.valueLink.value) || ""
-
-    this.props.valueLink.requestChange(
-      currentNumber ? currentNumber.format(this.props.format) : ""
-    )
+    return currentNumber ? currentNumber.format(this.props.format) : ""
   }
 
   _onBlur() {
-    this._formatNumber()
+    let value = this.props.valueLink.value
+    value = value.replace(/,/g, "")
+    value = this._toNumeral(value) || ""
+
+    this.props.valueLink.requestChange(value)
```

Comment:
Remove this line since the number is already being saved because of ```this.props,valueLink.requestChange``` is already being called as the user is entering the number.

If this line reminder the data will be saved twice once by ```this.props,valueLink.requestChange``` and another again once this component loss its force.